crates/tx3-lang/src/tx3.pest

```diff
@@ -165,9 +167,9 @@ input_block_from = { "from" ~ ":" ~ address_expr }
 input_block_datum_is = { "datum_is" ~ ":" ~ type }
 input_block_min_amount = { "min_amount" ~ ":" ~ asset_expr }
 input_block_redeemer = { "redeemer" ~ ":" ~ data_expr }
-input_block_ref = { "ref" ~ ":" ~ data_expr }
+input_block_ref = { "ref" ~ ":" ~ utxo_ref }
```

Comment:
Here we should keep data_expr as the value for the field. This gives more flexibility to the developer. For example, using a parameter as the source for the ref instead of a hardcoded value.

The analyzer should check that expression resolves to an utxo_ref and the compiler should coerce the expression into a utxo_ref.